router.go

```diff
@@ -168,12 +180,13 @@ func regHandler(w http.ResponseWriter, r *http.Request) {
 		queryString = "UPDATE temp_users SET token = $1 WHERE phone_number = $2"
 		stmt, err = db.Prepare(queryString)
 		res, err := stmt.Exec(token, newUser.PhoneNumber)
-		failOnError(err, "Problem with update query")
+		if err != nil{
+			failWithStatusCode(err, "Problem with update query", w, http.StatusInternalServerError)
+			return
+		}
 		numRows, err := res.RowsAffected()
 		if numRows < 1 {
-			failOnError(err, "Unable to update new user")
-			w.WriteHeader(http.StatusConflict)
-			fmt.Fprintf(w, "Server Error")
+			failWithStatusCode(err, "Unable to update user", w, http.StatusInternalServerError)
```

Comment:
numRows could be less than one, with with err being nil (ie, the update query runs successfully but does not affect any rows, and rowsAffected is called successfully and returns 0), but this won't send any response in that case.